### PR TITLE
feat: add expand/collapse icon to menu items

### PR DESCRIPTION
**Pull request by scomea**
# Description
Ads expand and collapse icons to menu items.

## Motivation & context
An expected feature of nested menus.

## Issue type checklist
- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [x] I have modified an existing component
- [x] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-foundation/src/menu-item/menu-item.ts [hunk 1]
```diff
@@ -42,14 +42,12 @@ export class MenuItem extends FASTElement {
...
-            const submenu: Element | undefined = this.getSubmenu();
-            if (submenu === undefined) {
+            if (this.submenu === undefined) {
                 return;
             }
             if (this.expanded === false) {
-                (submenu as Menu).collapseExpandedItem();
+                (this.submenu as Menu).collapseExpandedItem();
             } else {
                 this.currentDirection = getDirection(this);
```
Comment (robarbms) [inline packages/web-components/fast-foundation/src/menu-item/menu-item.ts:52]: I haven't used getDirection so I am guessing at the purpose here. Is it to inherit direction from the host menu component? From a performance standpoint it makes sense to only set it if the submenu is open. But with the indicator having RTL specific styles, would this need to be called in connectedCallback now?

**Review by robarbms — CHANGES_REQUESTED** [review]

File: packages/web-components/fast-foundation/src/menu-item/menu-item.ts [hunk 2]
```diff
@@ -83,17 +81,18 @@ export class MenuItem extends FASTElement {
     }
 
     /**
+     * reference to the anchored region
+     *
      * @internal
      */
     @observable
-    public submenu: Element | undefined;
+    public submenuRegion: AnchoredRegion;
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/menu-item/menu-item.ts:89]: @scomea is this a breaking change?

Comment (scomea) [reply packages/web-components/fast-foundation/src/menu-item/menu-item.ts:89]: It's marked internal so I'd say that means folks shouldn't be taking dependencies on it, but it is an unnecessary change so I'll revert it to minimize churn

File: packages/web-components/fast-foundation/src/menu-item/menu-item.ts (hunk 1 above)
Comment (scomea) [reply packages/web-components/fast-foundation/src/menu-item/menu-item.ts:52]: No, we get is specifically to populate the dir attribute of the anchored region when it is created.  All the css layout stuff just works.

File: packages/web-components/fast-foundation/src/menu-item/menu-item.ts [hunk 3]
```diff
@@ -254,11 +259,13 @@ export class MenuItem extends FASTElement {
      *
      * @internal
      */
-    private getSubmenu = (): undefined | Element => {
+    private updateSubmenu = (): void => {
         const domChildren: Element[] = Array.from(this.children);
```
[the remark this answers is not in the document]
Comment (scomea) [reply packages/web-components/fast-foundation/src/menu-item/menu-item.ts]: yep

**Review by chrisdholt — APPROVED** [review]

**Review by robarbms — APPROVED** [review]

**Pull request merged**